src/main/java/org/killbill/billing/plugin/adyen/client/model/PurchaseResult.java

```diff
@@ -118,21 +121,26 @@ public String getResultCode() {
         return errorCodes;
     }
 
+    public Optional<AdyenCallErrorStatus> getAdyenCallErrorStatus() {
```

Comment:
yeah because that would also be the next bigger refactoring. And you were already not amused about the amount of changes lines ;). As said the plugin is in a pretty bad shape so the cases should be handled consistent but it's not (yet) the case. 

Next point for a follow up PR